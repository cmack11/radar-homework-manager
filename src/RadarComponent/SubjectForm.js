import React from 'react';
import { connect } from 'react-redux';
import { retrieveAssignments, addSubject } from '../actions/assignmentAction.js';

const mapDispatchToProps = dispatch => ({
 retrieveAssignments: () => dispatch(retrieveAssignments()),
 addSubject: (subject) => dispatch(addSubject(subject))
})

const mapStateToProps = state => {
    console.log("Map :"+ JSON.stringify(state));
    return {
      /* map this later
      name:"",
      color:"",
      assignments:[],
      */
    }
  }

const maxNameLength = 14;
const maxSubjectLength = 8;

export class SubjectForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = this.getDefaultState();

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event) {
    const target = event.target;
    const name = target.name;
    let value = target.value;

    this.setState({
      [name]: value
    });
  }

  handleSubmit(event) {
    //alert('The following Subject was submitted: ' + this.state.subjectName);
    let colors = ['red','blue','green','orange','black','magenta','maroon','cyan'];
    let index = Math.floor(Math.random() * colors.length);
    let color = colors[index];

    if(!this.allValid()) return; 

    this.props.addSubject(
      {name:this.state.subjectName, color:color, assignments:[], description:this.state.subjectDesc, defaultType:this.state.defaultTaskType});

    this.setState(this.getDefaultState())
    if(this.props.closeForm)
      this.props.closeForm();
  }

  allValid() {
    if(this.state.subjectName && this.state.subjectName.length > maxNameLength)
      return false;
    if(this.props.subjectNames) {
      if(this.props.subjectNames.length === maxSubjectLength) return false;
      let duplicate = false;
      duplicate = this.props.subjectNames.some((name)=>{
        return name.toLowerCase() === this.state.subjectName.toLowerCase();
      })
      if(duplicate) return false;
    }


    return true;
  }

  getDefaultState() {
    return {subjectName:'', subjectDesc:'', defaultTaskType: 'Assignment'};
  }

  render() {

    let taskTypeOptions = [];
    for (let i = 0; i < this.props.taskTypes.length; ++i)
    {
      const taskType = this.props.taskTypes[i];
      taskTypeOptions.push(<option value={taskType}>{taskType}</option>);
    }

    return (
    <div style={{position:'absolute',
          backgroundColor:'grey',
          width:this.props.width,
          right:(window.innerWidth-this.props.width)/2,
          bottom:(window.innerHeight-150)/2,
          padding:20
        }}>
      <form >
        <b className='form-fields' onClick={this.props.switchForm}>Add Subject</b>
        <label className='form-fields'>
          Subject Name:
          <input name="subjectName" type="text" value={this.state.subjectName} onChange={this.handleChange} />
        </label>
        <label className='form-fields'>
          Subject Description:
          <input name="subjectDesc" type="text" value={this.state.subjectDesc} onChange={this.handleChange} />
        </label>
        <label className='form-fields'>
          Default Task Type:
          <select name="defaultTaskType" value={this.state.defaultTaskType} onChange={this.handleChange}>
            {taskTypeOptions}
          </select>
        </label>
        <button type="button" value="Submit" onClick={this.handleSubmit}>Submit</button>
      </form>
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SubjectForm);
