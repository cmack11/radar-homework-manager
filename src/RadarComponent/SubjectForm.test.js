import React from 'react';
import {SubjectForm} from './SubjectForm.js';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import sinon from "sinon";

beforeAll(() => {
  Enzyme.configure({ adapter: new Adapter() })
  jest.useFakeTimers();
});

const event = {
	target: '',
	name : '',
	value: '',
	preventDefault: () => {},
}

describe('the handleChange funtion', () => {
	
	it('should call handleChange', () => {
  		const component = shallow(<SubjectForm subjectNames={[]} taskTypes={[]} addSubject={()=>{}}/>,);
		const spy = jest.spyOn(component.instance(),'handleChange');
  	
  		component.instance().handleChange(event);
  	
  		expect(spy).toBeCalled();
   });
})

describe('the handleSubmit funtion', () => {
	
	it('should call handleSubmit', () => {
  		const component = shallow(<SubjectForm subjectNames={[]} taskTypes={[]} addSubject={()=>{}}/>,);
		const spy = jest.spyOn(component.instance(),'handleSubmit');
  	
  		component.instance().handleSubmit(event);
  	
  		expect(spy).toBeCalled();
   });
})

describe('isValid Check', () => {
	
	it('Check if valid input gets submitted', () => {
		const wrapper = shallow(
			<SubjectForm taskTypes={["assignment", "exam"]} subjectNames = {["sub1", "sub2"]} addSubject={() => {}}/>
		);
		let instance = wrapper.instance();
		instance.state.subjectName = "NewSub";
		instance.state.subjectDesc = "SOME DESC";
		instance.state.defaultTaskType = "assignment";
		instance.handleSubmit();
		expect(wrapper).toMatchSnapshot();
   });
})

//   {name:this.state.subjectName, color:color, assignments:[], description:this.state.subjectDesc, defaultType:this.state.defaultTaskType});