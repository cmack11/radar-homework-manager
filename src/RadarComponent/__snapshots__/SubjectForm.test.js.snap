// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isValid Check Check if valid input gets submitted 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubjectForm
    addSubject={[Function]}
    subjectNames={
      Array [
        "sub1",
        "sub2",
      ]
    }
    taskTypes={
      Array [
        "assignment",
        "exam",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Form
        as="form"
      >
        <div
          className="subject-title-container"
        >
          <b>
            Add Subject
          </b>
          <div
            className="switch-icon"
          >
            <Context.Provider
              value={
                Object {
                  "size": 20,
                }
              }
            >
              <MdRepeat />
            </Context.Provider>
          </div>
        </div>
        <FormField
          className="form-fields"
        >
          <label
            className="label-text label-center"
          >
            Subject Name
          </label>
          <input
            name="subjectName"
            onChange={[Function]}
            style={
              Object {
                "borderColor": null,
              }
            }
            type="text"
            value=""
          />
        </FormField>
        <FormField
          className="form-fields"
        >
          <label
            className="label-text"
          >
            Subject Description
          </label>
          <input
            name="subjectDesc"
            onChange={[Function]}
            type="text"
            value=""
          />
        </FormField>
        <FormField
          className="form-fields"
        >
          <label
            className="label-text"
          >
            Default Task Type
          </label>
          <select
            name="defaultTaskType"
            onChange={[Function]}
            value="Assignment"
          >
            <option
              value="assignment"
            >
              assignment
            </option>
            <option
              value="exam"
            >
              exam
            </option>
          </select>
        </FormField>
        <Button
          as="button"
          onClick={[Function]}
          primary={true}
          role="button"
          type="button"
          value="Submit"
        >
          Submit
        </Button>
      </Form>,
      "className": "subject-form",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "as": "form",
        "children": Array [
          <div
            className="subject-title-container"
          >
            <b>
              Add Subject
            </b>
            <div
              className="switch-icon"
            >
              <Context.Provider
                value={
                  Object {
                    "size": 20,
                  }
                }
              >
                <MdRepeat />
              </Context.Provider>
            </div>
          </div>,
          <FormField
            className="form-fields"
          >
            <label
              className="label-text label-center"
            >
              Subject Name
            </label>
            <input
              name="subjectName"
              onChange={[Function]}
              style={
                Object {
                  "borderColor": null,
                }
              }
              type="text"
              value=""
            />
          </FormField>,
          <FormField
            className="form-fields"
          >
            <label
              className="label-text"
            >
              Subject Description
            </label>
            <input
              name="subjectDesc"
              onChange={[Function]}
              type="text"
              value=""
            />
          </FormField>,
          <FormField
            className="form-fields"
          >
            <label
              className="label-text"
            >
              Default Task Type
            </label>
            <select
              name="defaultTaskType"
              onChange={[Function]}
              value="Assignment"
            >
              <option
                value="assignment"
              >
                assignment
              </option>
              <option
                value="exam"
              >
                exam
              </option>
            </select>
          </FormField>,
          <Button
            as="button"
            onClick={[Function]}
            primary={true}
            role="button"
            type="button"
            value="Submit"
          >
            Submit
          </Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <b>
                Add Subject
              </b>,
              <div
                className="switch-icon"
              >
                <Context.Provider
                  value={
                    Object {
                      "size": 20,
                    }
                  }
                >
                  <MdRepeat />
                </Context.Provider>
              </div>,
            ],
            "className": "subject-title-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Subject",
              },
              "ref": null,
              "rendered": "Add Subject",
              "type": "b",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Context.Provider
                  value={
                    Object {
                      "size": 20,
                    }
                  }
                >
                  <MdRepeat />
                </Context.Provider>,
                "className": "switch-icon",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <MdRepeat />,
                  "value": Object {
                    "size": 20,
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": Object {
                  "$$typeof": Symbol(react.provider),
                  "_context": Object {
                    "$$typeof": Symbol(react.context),
                    "Consumer": [Circular],
                    "Provider": [Circular],
                    "_calculateChangedBits": null,
                    "_currentRenderer": null,
                    "_currentRenderer2": null,
                    "_currentValue": Object {
                      "attr": undefined,
                      "className": undefined,
                      "color": undefined,
                      "size": undefined,
                      "style": undefined,
                    },
                    "_currentValue2": Object {
                      "attr": undefined,
                      "className": undefined,
                      "color": undefined,
                      "size": undefined,
                      "style": undefined,
                    },
                    "unstable_read": [Function],
                  },
                },
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <label
                className="label-text label-center"
              >
                Subject Name
              </label>,
              <input
                name="subjectName"
                onChange={[Function]}
                style={
                  Object {
                    "borderColor": null,
                  }
                }
                type="text"
                value=""
              />,
            ],
            "className": "form-fields",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Subject Name",
                "className": "label-text label-center",
              },
              "ref": null,
              "rendered": "Subject Name",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "subjectName",
                "onChange": [Function],
                "style": Object {
                  "borderColor": null,
                },
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <label
                className="label-text"
              >
                Subject Description
              </label>,
              <input
                name="subjectDesc"
                onChange={[Function]}
                type="text"
                value=""
              />,
            ],
            "className": "form-fields",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Subject Description",
                "className": "label-text",
              },
              "ref": null,
              "rendered": "Subject Description",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "subjectDesc",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <label
                className="label-text"
              >
                Default Task Type
              </label>,
              <select
                name="defaultTaskType"
                onChange={[Function]}
                value="Assignment"
              >
                <option
                  value="assignment"
                >
                  assignment
                </option>
                <option
                  value="exam"
                >
                  exam
                </option>
              </select>,
            ],
            "className": "form-fields",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Default Task Type",
                "className": "label-text",
              },
              "ref": null,
              "rendered": "Default Task Type",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="assignment"
                  >
                    assignment
                  </option>,
                  <option
                    value="exam"
                  >
                    exam
                  </option>,
                ],
                "name": "defaultTaskType",
                "onChange": [Function],
                "value": "Assignment",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "assignment",
                    "value": "assignment",
                  },
                  "ref": null,
                  "rendered": "assignment",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "exam",
                    "value": "exam",
                  },
                  "ref": null,
                  "rendered": "exam",
                  "type": "option",
                },
              ],
              "type": "select",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": "Submit",
            "onClick": [Function],
            "primary": true,
            "role": "button",
            "type": "button",
            "value": "Submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Form
          as="form"
        >
          <div
            className="subject-title-container"
          >
            <b>
              Add Subject
            </b>
            <div
              className="switch-icon"
            >
              <Context.Provider
                value={
                  Object {
                    "size": 20,
                  }
                }
              >
                <MdRepeat />
              </Context.Provider>
            </div>
          </div>
          <FormField
            className="form-fields"
          >
            <label
              className="label-text label-center"
            >
              Subject Name
            </label>
            <input
              name="subjectName"
              onChange={[Function]}
              style={
                Object {
                  "borderColor": null,
                }
              }
              type="text"
              value=""
            />
          </FormField>
          <FormField
            className="form-fields"
          >
            <label
              className="label-text"
            >
              Subject Description
            </label>
            <input
              name="subjectDesc"
              onChange={[Function]}
              type="text"
              value=""
            />
          </FormField>
          <FormField
            className="form-fields"
          >
            <label
              className="label-text"
            >
              Default Task Type
            </label>
            <select
              name="defaultTaskType"
              onChange={[Function]}
              value="Assignment"
            >
              <option
                value="assignment"
              >
                assignment
              </option>
              <option
                value="exam"
              >
                exam
              </option>
            </select>
          </FormField>
          <Button
            as="button"
            onClick={[Function]}
            primary={true}
            role="button"
            type="button"
            value="Submit"
          >
            Submit
          </Button>
        </Form>,
        "className": "subject-form",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "form",
          "children": Array [
            <div
              className="subject-title-container"
            >
              <b>
                Add Subject
              </b>
              <div
                className="switch-icon"
              >
                <Context.Provider
                  value={
                    Object {
                      "size": 20,
                    }
                  }
                >
                  <MdRepeat />
                </Context.Provider>
              </div>
            </div>,
            <FormField
              className="form-fields"
            >
              <label
                className="label-text label-center"
              >
                Subject Name
              </label>
              <input
                name="subjectName"
                onChange={[Function]}
                style={
                  Object {
                    "borderColor": null,
                  }
                }
                type="text"
                value=""
              />
            </FormField>,
            <FormField
              className="form-fields"
            >
              <label
                className="label-text"
              >
                Subject Description
              </label>
              <input
                name="subjectDesc"
                onChange={[Function]}
                type="text"
                value=""
              />
            </FormField>,
            <FormField
              className="form-fields"
            >
              <label
                className="label-text"
              >
                Default Task Type
              </label>
              <select
                name="defaultTaskType"
                onChange={[Function]}
                value="Assignment"
              >
                <option
                  value="assignment"
                >
                  assignment
                </option>
                <option
                  value="exam"
                >
                  exam
                </option>
              </select>
            </FormField>,
            <Button
              as="button"
              onClick={[Function]}
              primary={true}
              role="button"
              type="button"
              value="Submit"
            >
              Submit
            </Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <b>
                  Add Subject
                </b>,
                <div
                  className="switch-icon"
                >
                  <Context.Provider
                    value={
                      Object {
                        "size": 20,
                      }
                    }
                  >
                    <MdRepeat />
                  </Context.Provider>
                </div>,
              ],
              "className": "subject-title-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Subject",
                },
                "ref": null,
                "rendered": "Add Subject",
                "type": "b",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Context.Provider
                    value={
                      Object {
                        "size": 20,
                      }
                    }
                  >
                    <MdRepeat />
                  </Context.Provider>,
                  "className": "switch-icon",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <MdRepeat />,
                    "value": Object {
                      "size": 20,
                    },
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": Object {
                    "$$typeof": Symbol(react.provider),
                    "_context": Object {
                      "$$typeof": Symbol(react.context),
                      "Consumer": [Circular],
                      "Provider": [Circular],
                      "_calculateChangedBits": null,
                      "_currentRenderer": null,
                      "_currentRenderer2": null,
                      "_currentValue": Object {
                        "attr": undefined,
                        "className": undefined,
                        "color": undefined,
                        "size": undefined,
                        "style": undefined,
                      },
                      "_currentValue2": Object {
                        "attr": undefined,
                        "className": undefined,
                        "color": undefined,
                        "size": undefined,
                        "style": undefined,
                      },
                      "unstable_read": [Function],
                    },
                  },
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <label
                  className="label-text label-center"
                >
                  Subject Name
                </label>,
                <input
                  name="subjectName"
                  onChange={[Function]}
                  style={
                    Object {
                      "borderColor": null,
                    }
                  }
                  type="text"
                  value=""
                />,
              ],
              "className": "form-fields",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Subject Name",
                  "className": "label-text label-center",
                },
                "ref": null,
                "rendered": "Subject Name",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "subjectName",
                  "onChange": [Function],
                  "style": Object {
                    "borderColor": null,
                  },
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <label
                  className="label-text"
                >
                  Subject Description
                </label>,
                <input
                  name="subjectDesc"
                  onChange={[Function]}
                  type="text"
                  value=""
                />,
              ],
              "className": "form-fields",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Subject Description",
                  "className": "label-text",
                },
                "ref": null,
                "rendered": "Subject Description",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "subjectDesc",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <label
                  className="label-text"
                >
                  Default Task Type
                </label>,
                <select
                  name="defaultTaskType"
                  onChange={[Function]}
                  value="Assignment"
                >
                  <option
                    value="assignment"
                  >
                    assignment
                  </option>
                  <option
                    value="exam"
                  >
                    exam
                  </option>
                </select>,
              ],
              "className": "form-fields",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Default Task Type",
                  "className": "label-text",
                },
                "ref": null,
                "rendered": "Default Task Type",
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <option
                      value="assignment"
                    >
                      assignment
                    </option>,
                    <option
                      value="exam"
                    >
                      exam
                    </option>,
                  ],
                  "name": "defaultTaskType",
                  "onChange": [Function],
                  "value": "Assignment",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "assignment",
                      "value": "assignment",
                    },
                    "ref": null,
                    "rendered": "assignment",
                    "type": "option",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "exam",
                      "value": "exam",
                    },
                    "ref": null,
                    "rendered": "exam",
                    "type": "option",
                  },
                ],
                "type": "select",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "children": "Submit",
              "onClick": [Function],
              "primary": true,
              "role": "button",
              "type": "button",
              "value": "Submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
